docs: add migration guide

dancormier opened the pull request:
Work in progress

dancormier committed:
docs: add migration guide

dancormier committed:
Remove some extra returns

giamir commented on MIGRATION_GUIDE.md:
```diff
@@ -0,0 +1,240 @@
+# Migrating from Stacks v1 to v2
+
+With the release of Stacks v2, we've made some breaking changes to the library. This purpose of this guide is to help you prepare your code be compatible with the changes in Stacks v2.
```
```suggestion
With the release of Stacks v2, we've made some breaking changes to the library. The purpose of this guide is to help you prepare your code be compatible with the changes in Stacks v2.
```

giamir commented on MIGRATION_GUIDE.md:
```diff
@@ -0,0 +1,240 @@
+
+## Upgrading dependencies
+
+<!-- TODO outline the need to update Stacks-Editor, Stacks-Icons, etc -->
```
I was reading in the markdown preview and I could not see this comment. Maybe it would be less confusing to leave the TODO in cursive so that people understand that there is not a mistake in this section but it is in fact a work in progress. 🙂

giamir commented on MIGRATION_GUIDE.md:
```diff
@@ -0,0 +1,240 @@
+
+## Custom theme generation
+
+Stacks v2 has deprecated the Less function `.generate-calculated-themed-variables()` in favor of the newly added `.create-custom-theme-hsl-variables()` function. This new function works differently behind the scenes, but it accepts the same arguments and generates the same CSS custom properties as it's deprecated equivelent. 
```
```suggestion
Stacks v2 has deprecated the Less function `.generate-calculated-themed-variables()` in favor of the newly added `.create-custom-theme-hsl-variables()` function. This new function works differently behind the scenes, but it accepts the same arguments and generates the same CSS custom properties as its deprecated equivelent. 
```

giamir commented on MIGRATION_GUIDE.md:
```diff
@@ -0,0 +1,240 @@
+
+To update to the new custom theme variable generation function, simply search for `.generate-calculated-themed-variables` in your codebase and replace it with `.create-custom-theme-hsl-variables`.
+
+## Other deprecations
```
Also for these last 2 sections we could put a WIP sign for the content since they look odd in the markdown preview (almost as if we forgot something).

giamir reviewed (approved):
@dancormier The draft looks good to me. I think there is enough information for (potentially) product engineers starting the integration on Monday. We can add more details to this document as the work progress. By the time we will merge `stacks-next` in `develop` it will be ready. 

The only part that could probably be completed already is the first TODO about Stacks Icons and Stacks Editor RCs going hand in hand with Stacks Classic v2. A short paragraph would be more than enough in my opinion.

Thanks for moving this forward. 🎉

dancormier committed:
Update MIGRATION_GUIDE.md

Co-authored-by: Giamir Buoncristiani <gbuoncristiani@stackoverflow.com>

dancormier committed:
Apply suggestions from code review

Co-authored-by: Giamir Buoncristiani <gbuoncristiani@stackoverflow.com>

dancormier committed:
Make theming alterations

dancormier committed:
Add "Upgrading dependencies" section

dancormier committed:
Minor tweaks to migration guide

dancormier committed:
Merge branch 'dcormier/stacks-next-migration-guide' of https://github.com/StackExchange/Stacks into dcormier/stacks-next-migration-guide

dancormier committed:
The definite article

☝️ good professional wrestler name. When others refer to you as "Definite Article", you could angrily correct them: "It's **The** Definite Article"

dancormier committed:
Update MIGRATION_GUIDE.md
